packages/eas-cli/src/commands/device/delete.ts

```diff
@@ -0,0 +1,184 @@
+import { Device, DeviceStatus, RequestContext } from '@expo/apple-utils';
+import { getConfig } from '@expo/config';
+import { Flags } from '@oclif/core';
+import assert from 'assert';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import { chooseDevicesToDeleteAsync } from '../../credentials/ios/actions/DeviceUtils';
+import { AppleDeviceMutation } from '../../credentials/ios/api/graphql/mutations/AppleDeviceMutation';
+import {
+  AppleDeviceQuery,
+  AppleDeviceQueryResult,
+} from '../../credentials/ios/api/graphql/queries/AppleDeviceQuery';
+import { AppleTeamQuery } from '../../credentials/ios/api/graphql/queries/AppleTeamQuery';
+import { authenticateAsync, getRequestContext } from '../../credentials/ios/appstore/authenticate';
+import formatDevice from '../../devices/utils/formatDevice';
+import { AppleDevice } from '../../graphql/generated';
+import Log from '../../log';
+import { ora } from '../../ora';
+import { findProjectRootAsync, getProjectAccountNameAsync } from '../../project/projectUtils';
+import { promptAsync, toggleConfirmAsync } from '../../prompts';
+
+export default class DeviceDelete extends EasCommand {
+  static description = 'remove a registered device from your account';
+
+  static flags = {
+    'apple-team-id': Flags.string(),
+    udid: Flags.string({ multiple: true }),
+  };
+
+  async runAsync(): Promise<void> {
```

Comment:
This function is 130 LOC. That's a lot. Please break it down.

Reply:
`list` is 100 LOC. What's the target value?

Reply:
Basically, I think we should have a lint rule for that. It's hard to add that in right now as it'd require a lot of refactoring. I'd say a rule of thumb is that a function should fit a reasonable display (without scrolling).

Reply:
Works for me, I'll make a note to include that rule for myself whenever I need to 'touch' existing code as well.